### DISCUSSION TITLE
Recommended way to put props into atoms

### DISCUSSION THREAD

**Discussion by DonaldTaylor** [pmndrs/jotai#1372] [Q&A]
What is the recommended way to get props into atoms?

My use case is that I have a library which takes props that I want to put into atoms for access across the various components of the library.

Currently, I do the following, which works fine—I use `useEffect` to update the atom when the prop changes. I'm just wondering if this is the recommended approach.

```
function MyLibrary({ data: dataProp }) {
  const [data, setData] = useAtom(dataAtom);
  useEffect(() => {
    setData(dataProp);
  }, [dataProp]);
  // . . .
}
```

Thank you.

**Comment by dai-shi** [comment]
The best is not to make atoms depend on props. This might not be possible with some libs that are out of your control. 

Three solutions would be:
- Provider's initialValues prop
- create a new atom from prop and possibly use atoms-in-atom

If none works, useEffect is an unavoidable solution.
